IFS-4980 Erstellung initiale Gliederung und Einleitung & Zielbild für Referenzarchitektur Frontend (#974)
IsyFact/IsyFact-Standards: isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/frontend.adoc

include::documentation-guide::partial$content-new.adoc[]

Die xref:glossary::terms-definitions.adoc#glossar-referenzarchitektur[Referenzarchitektur] Frontend beschreibt die grundlegenden Rahmenbedingungen sowie die architekturellen Leitlinien für die Entwicklung des xref:glossary::terms-definitions.adoc#glossar-it-systemtyp[IT-Systemtyps Frontend] in Anwendungen.
Sie dient als verbindliche Grundlage für die Gestaltung und Umsetzung benutzerorientierter Komponenten im Kontext moderner Geschäftsanwendungen.

[[fachliche-referenzarchitektur-frontend]]
.Einbettung Frontends in der fachlichen Referenzarchitektur
image::referenzarchitektur:software-technisch/frontend/kotext-referenzarchitektur-frontend.drawio.svg[width=50%,align=center]

> Wie in <<fachliche-referenzarchitektur-frontend>> dargestellt, erfolgt grundsätzlich eine Unterteilung der xref:glossary::terms-definitions.adoc#glossar-geschaeftsanwendung[Geschäftsanwendung (GA)] in zwei zentrale Bereiche:

Reviewer: ```suggestion
Wie in <<fachliche-referenzarchitektur-frontend>> dargestellt, erfolgt grundsätzlich eine Unterteilung der xref:glossary::terms-definitions.adoc#glossar-geschaeftsanwendung[Geschäftsanwendung (GA)] in zwei IT-Systeme:
```

> die Geschäftskomponente Backend (GK) zur Bereitstellung fachlicher Logik und Services sowie die Geschäftskomponente Frontend (GKGUI) als Benutzeroberfläche zur Interaktion mit dem Anwender.

Reviewer: Der Begriff Geschäftskomponente kommt in der Dokumentation aktuell nur (noch) in den Namenskonventionen vor. 
```suggestion
das Backend (GK) zur Bereitstellung fachlicher Logik und Services sowie das Frontend (GKGUI) als Benutzeroberfläche zur Interaktion mit dem Anwender.
```

Diese softwaretechnische Trennung erfolgt ausschließlich aus technologischen und organisatorischen Gründen: Frontend und Backend können unabhängig voneinander entwickelt, betrieben und skaliert werden und setzen in der Regel unterschiedliche Technologien sowie potenziell getrennte Entwicklungsteams ein.
Aus Sicht der Anwender bleibt die Anwendung dabei weiterhin als einheitliches System wahrnehmbar.

> Dieses Strukturprinzip gilt analog für xref:glossary::terms-definitions.adoc#glossar-querschnittsanwendung[Querschnittsanwendungen (QA)], die entsprechend in QK (Querschnittskomponente) und QKGUI (Querschnitts-Frontend) gegliedert werden.

Reviewer: ```suggestion
Dieses Strukturprinzip gilt analog für xref:glossary::terms-definitions.adoc#glossar-querschnittsanwendung[Querschnittsanwendungen (QA)], die entsprechend in QK (Backend) und QKGUI (Frontend) gegliedert werden.
```

image::referenzarchitektur:software-technisch/frontend/software-technisch-frontend.svg[width=50%,align=center]

Durch die Verwendung von SPAs befindet sich der Frontend-Teil der GA im Webbrowser des Anwenders, während der Backend-Teil weiterhin auf den Servern der Anwendungslandschaft betrieben wird.
Aus diesem Grund bietet es sich an, ein zentrales API-Gateway als querschnittliche Komponente zwischen Frontends und Backends zu schalten – insbesondere aus folgenden Gründen:

• Einheitliche Sicherheitskontrollen über alle APIs (Token-Validierung, Zugriffsschutz via Keycloak)
• Zentrales Routing zur Entkopplung von Frontends und Backends (Backend-Änderungen erfordern keine Anpassungen an SPAs)
• Konsistente technische Standards für alle Services (CORS, TLS, Protokolle, Header)
• Zentrale Durchsetzung von Rate Limiting, Timeouts und Circuit Breaking zum Schutz der Backends
• Vereinheitlichtes Monitoring, Logging und Tracing für vollständige Transparenz der Anfragen
• Unterstützung kontrollierter Deployment-Strategien wie API-Versionierung, Canary-Releases und Blue-Green-Deployments
• Vermeidung redundanter Security- und Infrastruktur-Implementierungen in einzelnen Services

> • Klare und konsistente Exposition interner und externer APIs

Reviewer: Der Teil passt m.E. besser an diese Stelle: https://isyfact.github.io/isyfact-standards-doku/dev/referenzarchitektur/software-technisch.html#api-gateway

Author: Die Inhalte wurden entsprechend verschoben.

image::referenzarchitektur:software-technisch/frontend/software-technisch-frontend.svg[width=50%,align=center]

Durch die Verwendung von SPAs befindet sich der Frontend-Teil der GA im Webbrowser des Anwenders, während der Backend-Teil weiterhin auf den Servern der Anwendungslandschaft betrieben wird.
Zur strukturierten, sicheren und einheitlichen Anbindung der Frontends an die Backend-Systeme ist verpflichtend ein zentrales API-Gateway als querschnittliche Komponente vorzuschalten.

> Das API-Gateway stellt einen eigenständigen Systemtyp dar und wird in der xref:referenzarchitektur:software-technisch.adoc[software-technischen Referenzarchitektur] detailliert beschrieben.

Reviewer: ```suggestion
Das API-Gateway ist ein Bestandteil der Systemlandschaft und wird in der xref:referenzarchitektur:software-technisch.adoc[software-technischen Referenzarchitektur] detailliert beschrieben.
```
